### PR TITLE
ModuleInterface: prepare to accept access-level on imports in swiftinterfaces

### PR DESCRIPTION
**Pull request by xymus**
This adds forward looking support for swiftinterfaces and the Access-levels on import features. Ensure the compiler accepts future swiftinterfaces using access-level keywords on imports without the flag. And print the `public` access-level keyword in Swift 6 mode.

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please smoke test

File: test/Sema/access-level-import-flag-check.swift [hunk 1]
```diff
@@ -28,3 +31,8 @@ package import PackageLib // expected-error@:1 {{Access level on imports require
 internal import InternalLib // expected-error@:1 {{Access level on imports require '-enable-experimental-feature AccessLevelOnImport'}}
 fileprivate import FileprivateLib // expected-error@:1 {{Access level on imports require '-enable-experimental-feature AccessLevelOnImport'}}
 private import PrivateLib // expected-error@:1 {{Access level on imports require '-enable-experimental-feature AccessLevelOnImport'}}
+
+//--- Client.swiftinterface
+// swift-interface-format-version: 1.0
+// swift-module-flags: -enable-library-evolution
+public import Swift
```
Comment (tshortli) [inline test/Sema/access-level-import-flag-check.swift:38]: Should we also test that other access levels are accepted, even though we do not plan to print them yet? I'm thinking of `package` in particular, which we do probably want to print in the future in some interfaces.

**Review by tshortli — APPROVED** [review]

**Comment by tshortli** [comment]
To minimize condfails I think it could be helpful to cherry pick this very targeted change back to `release/5.10`.

**Review by artemcm — APPROVED** [review]

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
@swift-ci Please smoke test

File: test/Sema/access-level-import-flag-check.swift (hunk 1 above)
Comment (xymus) [reply test/Sema/access-level-import-flag-check.swift]: Good idea, I've added them to the test.

**Review by tshortli — APPROVED** [review]

**Pull request merged**